modules/_node-scripts/tsc/index.mjs

```diff
@@ -18,11 +18,11 @@ import getFileProjectDir from '../util/getFileProjectDir.mjs';
 import getGitModifiedFiles from '../util/getGitModifiedFiles.mjs';
 import getNamedArguments from '../util/getNamedArguments.mjs';
 
-export default async function main() {
-	const {modifiedSince} = getNamedArguments({
-		modifiedSince: '--modified-since',
-	});
+const {modifiedSince} = getNamedArguments({
+	modifiedSince: '--modified-since',
+});
 
+export default async function main(sinceHash = modifiedSince) {
```

Comment:
I would also extract this logic to a different function in a different file with a proper name (eg: `runTscChecks(commit)`) and leave `index.mjs` with a content similar to your `checkTsc.mjs`. This is becuase this file is meant to be an entry point for a command so it's better not to use it as a general purpose library because it's mixing two responsibilities.

Also, related to my previous comment:

> Why running tsc if you are not going to return any errors? 🤔

If this logic was extracted to an utility function that returns the errors (an stdout+stderr buffer would be more than enough, there's no need parse `tsc`'s output) it could be used from both places and the output could be formatted as it is necessary in the calling context.